### PR TITLE
Improve types

### PR DESCRIPTION
**Pull request by t3chguy**
---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by florianduros — APPROVED** [review]
LGTM

**Review by andybalaam — CHANGES_REQUESTED** [review]

File: src/AddThreepid.ts [hunk 1]
```diff
@@ -256,10 +257,10 @@ export default class AddThreepid {
             } else {
                 await this.matrixClient.addThreePid(
                     {
-                        sid: this.sessionId,
+                        sid: this.sessionId!,
                         client_secret: this.clientSecret,
                         id_server: getIdServerDomain(this.matrixClient),
-                    },
+                    } as IBindThreePidBody,
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/AddThreepid.ts:263]: I agree with `Thee` but I think given this is a deprecated API which we have no sane means of testing holistically it'd be best to not touch it, Synapse can be touchy with `null` where it expects nothing

**Review by andybalaam — APPROVED** [review]

File: src/AddThreepid.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/AddThreepid.ts:263]: Then it doesn't match the spec. If anything we should have 2 different interfaces, one for the deprecated and one for stable

[the remark this answers is not in the document]
Comment (andybalaam) [reply src/AddThreepid.ts:263]: Certainly I don't think needing to match the spec is a good enough reason to use ` as IBindThreePidBody` which asserts something that isn't true.

[the remark this answers is not in the document]
Comment (t3chguy) [reply src/AddThreepid.ts]: Fixed

**Pull request merged**